src/ring_theory/fractional_ideal.lean

```diff
@@ -1048,6 +1048,27 @@ lemma mem_span_singleton_self (x : P) :
   x ∈ span_singleton S x :=
 (mem_span_singleton S).mpr ⟨1, one_smul _ _⟩
 
+lemma span_singleton_eq_span_singleton [no_zero_smul_divisors R P] {x y : P} (hx : x ≠ 0)
+  (hy : y ≠ 0) : span_singleton S x = span_singleton S y ↔ ∃ z : Rˣ, z • x = y :=
```

Comment:
Maybe it makes sense to use [mul_action.orbit_rel](https://leanprover-community.github.io/mathlib_docs/group_theory/group_action/basic.html#mul_action.orbit_rel) K[ˣ](https://leanprover-community.github.io/mathlib_docs/algebra/group/units.html#units) on the RHS. This looks very similar to the new `projectivization` stuff added in #12438. It looks like all you need to do here is glue together one of the proofs there with the fact that the coercion from `fractional_ideal` to `submodule` is injective.

Reply:
I can't see a nice way to do this, and I think it's overkill to import `projectivization` just to use something like `mk_eq_mk_iff`. I only need this particular form that mimics the existing statement for `mem_span_singleton`, so perhaps they can be refactored in another PR in the future to use `mul_action.orbit_rel`.

Reply:
Perhaps then you can use your new lemma to make the proof about `projectivization.submodule` shorter.

Reply:
The proofs in `projectivization` seem to be quite succinct already? I don't quite see which ones I should refactor using this.

Reply:
If they can be golfed, I think that shouldn't hold up this PR.

Reply:
I made the connection I was looking for in #12736